Project1/MoveHelper.h

```diff
@@ -1,40 +1,42 @@
 #pragma once
+
 #include <vector>
-#include "Map.h"
+#include "GridMap.h"
+#include "MapHelper.h"
+#include "reference_logic.h"
 
-class Map;
+class Unit;
 class CellPath;
 class GridCell;
-class Unit;
-class MoveHelper {
+class MoveHelper : 
+	public MapHelper 
+{
 private:
-	Map& _map;
 	template<typename T>
-	void vectorSubtract(std::vector<T>& a, std::vector<T>& b) {
+	void vectorSubtract(std::vector<std::reference_wrapper<T>>& a, std::vector<std::reference_wrapper<T>>& b) {
```

Comment:
this should be a free function

Reply:
Also, I couldn't find a one line set difference method in the STL to do this, so I made one...

Reply:
it also is free...